src/test/functional/steps/courts-search-and-sort.ts

```diff
@@ -0,0 +1,132 @@
+import {Then, When} from 'cucumber';
+import * as I from '../utlis/puppeteer.util';
+import {expect} from 'chai';
+
+When('I enter {string} into search textbox', async (searchText: string) => {
+  const selector = '#searchCourts';
+  const elementExist = await I.checkElement(selector);
+  expect(elementExist).equal(true);
+  await I.fillField(selector, searchText);
+});
+
+Then('All courts that include {string} should be displayed sorted by name', async (searchText: string) => {
+  const selector = '.govuk-table__cell.courtTableColumnName';
+  const courtHtmlElement: [string] = await I.getHtmlFromElements(selector);
+  expect(courtHtmlElement.length > 0).equal(true);
+
+  courtHtmlElement.forEach(courtName => expect(courtName.toLocaleLowerCase().includes(searchText)).equal(true));
+  const courtHtmlElementToSort: [string] = await I.getHtmlFromElements(selector);
+
+  const sortedCourtNames: [string] = courtHtmlElementToSort.sort();
+  let isEqual = true;
+  for (let i = 0; i < courtHtmlElement.length; ++i) {
+    isEqual = courtHtmlElement[i] === sortedCourtNames[i];
+  }
+  expect(isEqual).equal(true);
```

Comment:
This expectation only works for the last element in the above loop because it's over-written each time in the loop. To make it work like this, you'd need to set a variable in the loop only when the value is false.